app/src/main/java/at/team30/setroute/ui/route_detail/RouteDetailViewModel.kt

```diff
@@ -1,16 +1,63 @@
 package at.team30.setroute.ui.route_detail
 
+import android.app.Application
+import android.content.Context
+import android.graphics.Bitmap
+import android.util.Log
+import android.widget.ImageView
+import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.ViewModel
+import at.team30.setroute.BuildConfig
 import at.team30.setroute.infrastructure.IRoutesRepository
 import at.team30.setroute.models.Route
+import com.android.volley.Response
+import com.android.volley.toolbox.ImageRequest
+import com.android.volley.toolbox.Volley
 import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
 import javax.inject.Inject
 
 @HiltViewModel
 class RouteDetailViewModel @Inject constructor(
+        @ApplicationContext private val context : Context,
         private val routesRepository : IRoutesRepository
 ) : ViewModel() {
+
+
     fun getRoute(id: Int): Route? {
         return routesRepository.getRoutesById(id)
     }
+
+    fun requestImage(route: Route?,
+                     responseListener: Response.Listener<android.graphics.Bitmap>,
+                     errorListener: Response.ErrorListener) {
+        if (route == null) {
+            return;
+        }
+        val queue = Volley.newRequestQueue(context)
```

Comment:
Could move the queue to an private member of the ViewModel -> can be reused for other requests in the future.

Reply:
Sounds like a good idea :)